http/servlet-undertow/src/test/java/io/quarkus/ts/http/undertow/UndertowMissingServletClassIT.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.ts.http.undertow;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.bootstrap.Service;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnQuarkusVersion;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+@DisabledOnQuarkusVersion(version = "3\\.(8|9|10|11|12|13|14)\\..*|3\\.15\\.(0|1)(\\..*)?", reason = "Disabled on Quarkus versions before 3.15.2 due to build fail because missing servlet class configuration undertow issue https://github.com/quarkusio/quarkus/issues/43825")
+public class UndertowMissingServletClassIT {
+    @QuarkusApplication
+    static RestService app = new RestService()
+            .setAutoStart(false)
+            .onPreStart(UndertowMissingServletClassIT::replaceForInvalidXML);
+
+    @Tag("https://github.com/quarkusio/quarkus/issues/44063")
+    @Test
+    void verifyUndertowIgnoreServletClassMissing() {
+        assertDoesNotThrow(() -> app.start(),
+                "The app should start without any issue");
+        app.logs().assertDoesNotContain("Local name must not be null");
+    }
+
+    private static void replaceForInvalidXML(Service service) {
+        if (service instanceof RestService) {
+            RestService restService = (RestService) service;
+            Path invalidWebXml = Path.of("src/test/resources/META-INF/invalid-web.xml");
+
+            Path targetWebXml = restService.getServiceFolder().resolve("src/main/resources/META-INF/web.xml");
```

Comment:
I think you should copy it to jar itself, as it's already build if I'm not mistaken.

Reply:
Or you can just copy it to the service folder on prestart and force custom build. I added option for that with https://github.com/quarkus-qe/quarkus-test-framework/wiki/1.-Getting-Started#standalone-reproducer, you may need to rename `app` if you want to trigger this in the test properties. I suppose you can also put it to the `withProperty`, don't know, never tried that